src/main/java/net/dv8tion/jda/core/entities/impl/MessageImpl.java

```diff
@@ -142,7 +142,8 @@ else if (reaction.isSelf())
     public RestAction<Void> clearReactions()
     {
         checkPermission(Permission.MESSAGE_MANAGE);
-        return new RestAction<Void>(getJDA(), Route.Messages.REMOVE_ALL_REACTIONS.compile(getChannel().getId(), getId()), null)
+        Route.CompiledRoute route = Route.Messages.REMOVE_ALL_REACTIONS.compile(getChannel().getId(), getId());
+        return new RestAction<Void>(getJDA(), route, null)
```

Comment:
This method should use TextChannel#clearReactions and throw an exception if this message was not sent in a TextChannel